### DISCUSSION TITLE
How to display Many-To-Many relationship.

### DISCUSSION THREAD

**Discussion by sys-auditing** [yajra/laravel-datatables#2552] [Q&A]
### Summary of problem or feature request

Hi, 
i have Many-To-Many relationship between users & contacts table, so i would like to retrieve nested data for a logged in user (``` $request->user('api')->id ```).
As you know, this eager method ```with(['contacts.complaints' ``` returns a collection, that why i have this figure.

![Screenshot_1](https://user-images.githubusercontent.com/75130090/102479434-2f9fba80-405f-11eb-8599-5e2810d3aecc.png)
You can see here, this user is linked to 2 contacts, but are displayed in one row, i need that each result display in separate row.
 
So how can i loop over ``` contacts.complaints ``` and apply ``` $q->with(['comments.user', 'comments.replies.user']) ``` in order to display for each contact a correspondant complaints, comments ... in row.

Any help would be greatly appreciated

### Code snippet of problem

```php

//Contact Model
public function users()
{
    return $this->belongsToMany(User::class, 'user_contact')->withTimestamps();
}

//User Model
public function contacts()
{
    return $this->belongsToMany(Contact::class, 'user_contact')->withTimestamps();
} 

//Controller
 $complaints = User::whereId($request->user('api')->id)->with(['contacts.complaints' => function ($q)  {
               $q->with(['comments.user', 'comments.replies.user'])
                   ->latest();
               }])->get();
```
```js
//datatables js
columns: [
                {data: 'contacts[].account_number', name: 'contacts[].account_number'},
                {data: 'contacts[].email', name: 'contacts[].email'},
                {data: 'contacts[].name', name: 'contacts[].name'},
                {data: 'rate', name: 'rate'},
                {data: 'created_at', name: 'created_at'},
                {data: 'action', name: 'action', orderable: false, searchable: false},
            ],
```

### System details

- Operating System WSL2 Ubuntu 20.04
- PHP Version 7.4.3
- Laravel Version 8.10.0
- Laravel-Datatables Version v1.5.0

**Comment by yajra** [comment]
You can use [editColumn](https://yajrabox.com/docs/laravel-datatables/master/edit-column) or [addColumn](https://yajrabox.com/docs/laravel-datatables/master/add-column) then use the computed key as data for the multiple values.

Or you can also use the column renderer if you prefer js.

```php
// PHP
->addColumns('contacts_accounts', function($user) {
  return $user->contacts->implode('account_number', ', ');
})

// JS
{data: 'contacts_accounts', name: 'contacts[].account_number'},
```
